docs/design/seccomp.md

```diff
@@ -0,0 +1,364 @@
+
+Administrators should be able to control two aspects of defaults for seccomp
+profiles:
+
+1.  Defaults for classes of users based on pod security policy
+2.  Defaults at the API level
+
+#### Classes of users
+
+The [pod security policy](https://github.com/kubernetes/kubernetes/pull/7893)
+API extension governs the ability of users to make requests that affect pod
+and container security contexts.  The proposed design should deal with
+required changes to control access to new API features added.
+
+#### API Defaults
+
+There should be a minimum API-level guarantee for the seccomp profile to use
+when the field is not set when a pod is admitted to the API server.  This
+profile should be exposed as an argument to the API server.
+
+### Use Case: Seccomp profiles similar to container runtime defaults
+
+Many users will want to use images that make assumptions about running in the
+context of their chosen container runtime.  Such images are likely to
+frequently assume that they are running in the context of the container
+runtime's default seccomp settings.  Therefore, it should be possible to
+express a seccomp profile similar to a container runtime's defaults.
+
+### Use Case: Applications that link to libseccomp
+
+Some applications already link to libseccomp and control seccomp directly.  It
+should be possible to run these applications unmodified in Kubernetes; this
+implies there should be a way to disable seccomp control in Kubernetes for
+certain containers, or to run with a "no-op" or "unconfined" profile.
+
+### Use Case: Custom profiles
+
+Different applications have different requirements for seccomp profiles; it
+should be possible to express arbitrary seccomp profiles and use them in
+containers.
+
+## Community Work
+
+### Container runtime support for seccomp
+
+#### Docker / opencontainers
+
+Docker supports the open container initiative's API for
+seccomp, which is very close to the libseccomp API.  It allows full
+specification of seccomp filters, with arguments, operators, and actions.
+
+Docker allows the specification of a single seccomp filter.  There are
+community requests for:
+
+Issues:
+
+* [docker/22109](https://github.com/docker/docker/issues/22109): composable
+  seccomp filters
+* [docker/21105](https://github.com/docker/docker/issues/22109): custom
+  seccomp filters for builds
+
+#### rkt / appcontainers
+
+The `rkt` runtime delegates to systemd for seccomp support; there is an open
+issue to add support once `appc` supports it.  The `appc` project has an open
+issue to be able to describe seccomp as an isolator in an appc pod.
+
+The systemd seccomp facility is based on a whitelist of system calls that can
+be made, rather than a full filter specification.
+
+Issues:
+
+* [appc/529](https://github.com/appc/spec/issues/529)
+* [rkt/1614](https://github.com/coreos/rkt/issues/1614)
+
+#### HyperContainer
+
+[Hyper](https://hyper.sh) uses a hypervisor to run containers, so seccomp is
```

Comment:
fwiw, for rkt's LKVM stage, which uses a hypervisor to run pods, seccomp is still useful _within_ the pod to isolate individual containers from one another. I'm not sure what the deal is with Hyper (IIRC when it first came out it also ran pods within the VM, but most of their pod documentation is now 404ing and it seems like they now might be positioned entirely around containers instead)